src/components/pages/teams/TeamsForm.js

```diff
@@ -0,0 +1,168 @@
+    this.resetSelection(this.props.team);
+    this.setEditing(false) 
+ }
+
+  onInputChange = (e) => {
+    this.form[e.target.name] = e.target.value;
+  }
+
+  onAreaChange = (selected) => {
+    this.setState({selectedAreas: selected});
+    this.form.areas = selected.split(',');
+  }
+
+  handleFormUpdate = (field, value) => {
+    this.form[field] = value;
+  }
+
+  updateSelectedMembers = (selectedMembers, role) => {
+    switch (role) {
+      case MANAGER:
+        this.setState({ selectedManagers: selectedMembers });
+        this.form.managers = selectedMembers;    
+        break;
+      case CONFIRMED_USER:
+        this.setState({ selectedConfirmedUsers: selectedMembers });
+        this.form.confirmedUsers = selectedMembers;    
+        break;
+      case USER:
+        this.setState({ selectedUsers: selectedMembers });
+        this.form.users = selectedMembers;    
+        break;
+      default:
+        break;
+    }
+  }
+
+  render() {
+    const { team, areaValues } = this.props;
+    const title = this.props.team ? "teams.edit" : "teams.create";
+    return (
+      <div>
+        <Hero
+          title={title}
+        />
+        <Form onSubmit={this.handleSubmit}>
+          <div className="c-form">
+            <div className="row -main">
+              <div className="small-6 columns">
+                <div className="input-group">
+                  <label className="text"><FormattedMessage id={"teams.teamName"} /></label>
+                  <Input
+                    type="text"
+                    onChange={this.onInputChange}
+                    name="name"
+                    value={this.form.name || ''}
+                    placeholder={"Team name"}
+                    validations={['required']}
+                  />
+                </div>
+                <div className="input-group">
+                  <label className="text"><FormattedMessage id={"teams.areas"} /></label>
+                  <Select
+                    multi
+                    simpleValue
+                    name="areas-select"
+                    options={areaValues}
+                    value={this.state.selectedAreas}
+                    onChange={this.onAreaChange}
+                    searchable={false}
+                  />
+                </div>
+              </div>
+              <MembersManager 
+                updateSelectedMembers={this.updateSelectedMembers}
+                selectedUsers={this.state.selectedUsers}
+                selectedConfirmedUsers={this.state.selectedConfirmedUsers}
+                selectedManagers={this.state.selectedManagers}
+                addEmail={this.props.addEmail}
+              />
+            <div className="row small-12 columns">
+              <div className="c-form -nav">
+                { team &&
+                  <div><button type="button" onClick={this.handleCancel} className="c-button -light"><FormattedMessage id={"common.cancel"} /></button></div>}
```

Comment:
Can we set this to disabled when this.props.saving === true ? Incase the api takes its time.